R/foverlaps.R

```diff
@@ -54,7 +54,8 @@ foverlaps <- function(x, y, by.x=if (!is.null(key(x))) key(x) else key(y), by.y=
   yval1 = y[[yintervals[1L]]]; yval2 = y[[yintervals[2L]]]
   if (!storage.mode(xval1) %chin% c("double", "integer") || !storage.mode(xval2) %chin% c("double", "integer") || is.factor(xval1) || is.factor(xval2)) # adding factors to the bunch, #2645
     stop("The last two columns in by.x should correspond to the 'start' and 'end' intervals in data.table 'x' and must be integer/numeric type.")
-  isTRUEorNA <- function(x) isTRUE(x) || is.na(x) # x is assumed to  be logical
+  isTRUEorNA <- function(x) identical(x,TRUE) || identical(x,NA)  # also ensures x must be logical length 1
+  # base::isTRUE in 3.5.2 is implemented as `is.logical(x) && length(x)==1L && !is.na(x) && x` which seems inefficient vs just identical()
```

Comment:
minor difference is that `isTRUE` still returns true for named TRUE, while `identical` will return FALSE in such case.